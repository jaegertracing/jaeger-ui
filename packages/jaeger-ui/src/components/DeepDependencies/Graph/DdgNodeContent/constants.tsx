// Copyright (c) 2019 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import { getUrl as getSearchUrl } from '../../../SearchTracePage/url';

// Constants for calculating circular node positioning
export const FONT_SIZE = 14;
export const FONT = `${FONT_SIZE}px Helvetica Nueue`;
export const LINE_HEIGHT = 1.5;
export const OP_PADDING_TOP = 5;
export const PROGRESS_BAR_STROKE_WIDTH = 15;
export const RADIUS = 75;
export const WORD_RX = /\W*\w+\W*/g;

// While browsers suport URLs of unlimited length, many server clients do not handle more than this max
// export const MAX_LENGTH = 2083;
// get unpreditable, unreproducable results past 7400 characters
export const MAX_LENGTH = 7000;
export const MAX_LINKED_TRACES = 35;
export const MIN_LENGTH = getSearchUrl().length;
export const PARAM_NAME_LENGTH = '&traceID='.length;
