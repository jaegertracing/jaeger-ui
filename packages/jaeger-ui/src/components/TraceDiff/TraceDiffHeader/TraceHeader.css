/*
Copyright (c) 2017 Uber Technologies, Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

.TraecDiffHeader--traceHeader {
  align-items: stretch;
  background: #fafafa;
  border-right: 1px solid #d8d8d8;
  display: flex;
  flex-direction: column;
  flex: 1;
}

.TraecDiffHeader--traceTitle {
  align-items: center;
  background: #fff;
  border-bottom: 1px solid #eee;
  display: flex;
  flex: 1;
  font-size: 1.8em;
  margin: 0;
  padding: 0.25rem 2.5rem 0.25rem 1.25rem;
  position: relative;
}

.TraecDiffHeader--traceTitle.is-error {
  color: #c00;
}

.TraecDiffHeader--traceTitleChevron {
  color: var(--tx-color-title);
  font-size: 0.75em;
  margin-right: 0.75em;
  position: absolute;
  right: 0;
  top: calc(50% - 0.5em);
}

.TraecDiffHeader--traceAttributes {
  list-style: none;
  margin: 0;
  padding: 0.25rem 1.25rem;
}

.TraecDiffHeader--traceAttr {
  display: inline-block;
  margin-right: 1rem;
}
