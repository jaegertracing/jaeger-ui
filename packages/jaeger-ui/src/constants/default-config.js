// Copyright (c) 2017 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

export const FALLBACK_DAG_MAX_NUM_SERVICES = 100;
export const FALLBACK_TRACE_NAME_REGEXP_ALLOWLIST = [];

export const deprecations = [
  {
    formerKey: 'dependencies.dagMaxNumServices',
    currentKey: 'dependencies.menuEnabled',
    formerValue: 100,
    currentValue: false,
    callback: (config, formerKey, currentKey, formerValue, currentValue) => {
      const dagMaxNumServices = config.dependencies && config.dependencies.dagMaxNumServices;
      if (dagMaxNumServices !== FALLBACK_DAG_MAX_NUM_SERVICES) {
        // set the current key to true because the former value was not the default
        if (config.dependencies) {
          config.dependencies.menuEnabled = true;
        } else {
          config.dependencies = { menuEnabled: true };
        }
        // eslint-disable-next-line no-console
        console.warn(
          `Deprecated configuration key (${formerKey}: ${dagMaxNumServices}) detected, ` +
            `please use (${currentKey}: true) instead.`
        );
      }
    },
  },
];

export default {
  // Setting archiveEnabled to null means:
  // 1. Respect backend capability for non-memory storage
  // 2. Disable for in-memory storage (handled in getArchivedEnabled function)
  archiveEnabled: null,
  dependencies: {
    dagMaxNumServices: FALLBACK_DAG_MAX_NUM_SERVICES,
    menuEnabled: true,
  },
  search: {
    maxLookback: {
      label: '2 Days',
      value: '2d',
    },
    maxLimit: 1500,
  },
  tracking: {
    gaID: null,
    trackErrors: true,
    customWebAnalytics: null,
  },
  linkPatterns: [],
  menu: [
    {
      label: 'About Jaeger',
      items: [
        {
          label: 'GitHub',
          url: 'https://github.com/jaegertracing/jaeger',
        },
        {
          label: 'Docs',
          url: 'https://www.jaegertracing.io/docs/latest',
        },
        {
          label: 'Twitter',
          url: 'https://twitter.com/JaegerTracing',
        },
        {
          label: 'Discussion Group',
          url: 'https://groups.google.com/forum/#!forum/jaeger-tracing',
        },
        {
          label: 'Gitter.im',
          url: 'https://gitter.im/jaegertracing/Lobby',
        },
        {
          label: 'Blog',
          url: 'https://medium.com/jaegertracing',
        },
      ],
    },
  ],
  traceNameRegExpAllowList: FALLBACK_TRACE_NAME_REGEXP_ALLOWLIST,
};