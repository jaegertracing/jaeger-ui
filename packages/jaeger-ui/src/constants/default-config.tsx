// Copyright (c) 2017 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import deepFreeze from 'deep-freeze';

import { FALLBACK_DAG_MAX_NUM_SERVICES } from './index';

export default deepFreeze(
  Object.defineProperty(
    {
      archiveEnabled: false,
      dependencies: {
        dagMaxNumServices: FALLBACK_DAG_MAX_NUM_SERVICES,
        menuEnabled: true,
      },
      linkPatterns: [],
      menu: [
        {
          label: 'About Jaeger',
          items: [
            {
              label: 'GitHub',
              url: 'https://github.com/uber/jaeger',
            },
            {
              label: 'Docs',
              url: 'http://jaeger.readthedocs.io/en/latest/',
            },
            {
              label: 'Twitter',
              url: 'https://twitter.com/JaegerTracing',
            },
            {
              label: 'Discussion Group',
              url: 'https://groups.google.com/forum/#!forum/jaeger-tracing',
            },
            {
              label: 'Gitter.im',
              url: 'https://gitter.im/jaegertracing/Lobby',
            },
            {
              label: 'Blog',
              url: 'https://medium.com/jaegertracing/',
            },
          ],
        },
      ],
      search: {
        maxLookback: {
          label: '2 Days',
          value: '2d',
        },
        maxLimit: 1500,
      },
      tracking: {
        gaID: null,
        trackErrors: true,
      },
    },
    // fields that should be individually merged vs wholesale replaced
    '__mergeFields',
    { value: ['dependencies', 'search', 'tracking'] }
  )
);

export const deprecations = [
  {
    formerKey: 'dependenciesMenuEnabled',
    currentKey: 'dependencies.menuEnabled',
  },
  {
    formerKey: 'gaTrackingID',
    currentKey: 'tracking.gaID',
  },
];
