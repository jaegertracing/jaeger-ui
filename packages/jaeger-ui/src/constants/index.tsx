// Copyright (c) 2017 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

export const TOP_NAV_HEIGHT = 47 as 47;

export const FALLBACK_DAG_MAX_NUM_SERVICES = 100 as 100;
export const FALLBACK_TRACE_NAME = '<trace-without-root-span>' as '<trace-without-root-span>';
export const FALLBACK_STANDARD_DATETIME_FORMAT = 'LLL';

export const FETCH_DONE = 'FETCH_DONE' as 'FETCH_DONE';
export const FETCH_ERROR = 'FETCH_ERROR' as 'FETCH_ERROR';
export const FETCH_LOADING = 'FETCH_LOADING' as 'FETCH_LOADING';

export const fetchedState = {
  DONE: FETCH_DONE,
  ERROR: FETCH_ERROR,
  LOADING: FETCH_LOADING,
};
