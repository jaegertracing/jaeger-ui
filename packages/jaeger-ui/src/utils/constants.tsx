// Copyright (c) 2023 The Jaeger Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

/**
 * Provides access to constants injected by the build system.
 */

/**
 * Get the current execution environment, as inferred from NODE_ENV at build time.
 */
export function getAppEnvironment() {
  return __APP_ENVIRONMENT__;
}

/**
 * Get injected version details as a JSON-formatted string.
 */
export function getVersionInfo() {
  return __REACT_APP_VSN_STATE__;
}

export function shouldDebugGoogleAnalytics() {
  return __REACT_APP_GA_DEBUG__;
}
