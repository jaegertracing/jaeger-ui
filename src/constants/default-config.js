// Copyright (c) 2017 Uber Technologies, Inc.
//
// Permission is hereby granted, free of charge, to any person obtaining a copy
// of this software and associated documentation files (the "Software"), to deal
// in the Software without restriction, including without limitation the rights
// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
// copies of the Software, and to permit persons to whom the Software is
// furnished to do so, subject to the following conditions:
//
// The above copyright notice and this permission notice shall be included in
// all copies or substantial portions of the Software.
//
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
// THE SOFTWARE.

export default {
  menu: [
    {
      label: 'About Jaeger',
      items: [
        {
          label: 'GitHub',
          url: 'https://github.com/uber/jaeger',
        },
        {
          label: 'Docs',
          url: 'http://jaeger.readthedocs.io/en/latest/',
        },
        {
          label: 'Twitter',
          url: 'https://twitter.com/JaegerTracing',
        },
        {
          label: 'Discussion Group',
          url: 'https://groups.google.com/forum/#!forum/jaeger-tracing',
        },
        {
          label: 'Gitter.im',
          url: 'https://gitter.im/jaegertracing/Lobby',
        },
        {
          label: 'Blog',
          url: 'https://medium.com/jaegertracing/',
        },
      ],
    },
  ],
};
