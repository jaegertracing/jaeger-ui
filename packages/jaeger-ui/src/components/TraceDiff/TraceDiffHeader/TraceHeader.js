// @flow

// Copyright (c) 2017 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import * as React from 'react';
import IoChevronDown from 'react-icons/lib/io/chevron-down';

import TraceTimelineLink from './TraceTimelineLink';
import RelativeDate from '../../common/RelativeDate';
import TraceName from '../../common/TraceName';
import { fetchedState } from '../../../constants';
import { formatDuration } from '../../../utils/date';

import type { FetchedState } from '../../../types';
import type { ApiError } from '../../../types/api-error';

import './TraceHeader.css';

type Props = {
  duration: ?number,
  error?: ApiError,
  startTime: ?number,
  state: ?FetchedState,
  traceID: ?string,
  traceName: ?string,
  totalSpans: ?number,
};

type AttrsProps = {
  startTime: ?number,
  duration: ?number,
  totalSpans: ?number,
};

// exported for tests
export function EmptyAttrs() {
  return (
    <ul className="TraecDiffHeader--traceAttributes">
      <li className="TraecDiffHeader--traceAttr">&nbsp;</li>
    </ul>
  );
}

// exported for tests
export function Attrs(props: AttrsProps) {
  const { startTime, duration, totalSpans } = props;
  return (
    <ul className="TraecDiffHeader--traceAttributes">
      <li className="TraecDiffHeader--traceAttr">
        <strong>
          <RelativeDate value={(startTime || 0) / 1000} includeTime fullMonthName />
        </strong>
      </li>
      <li className="TraecDiffHeader--traceAttr">
        <span className="u-tx-muted">Duration: </span>
        <strong>{formatDuration(duration || 0)}</strong>
      </li>
      <li className="TraecDiffHeader--traceAttr">
        <span className="u-tx-muted">Spans: </span> <strong>{totalSpans || 0}</strong>
      </li>
    </ul>
  );
}

export default function TraceHeader(props: Props) {
  const { duration, error, startTime, state, traceID, totalSpans, traceName } = props;
  const AttrsComponent = state === fetchedState.DONE ? Attrs : EmptyAttrs;
  return (
    <div className="TraecDiffHeader--traceHeader">
      <h1 className="TraecDiffHeader--traceTitle">
        <span>
          {traceID ? (
            <React.Fragment>
              <TraceName key="name" traceName={traceName} error={error} state={state} />{' '}
              <small key="id" className="u-tx-muted ub-pr2">
                {traceID.slice(0, 7)}
              </small>
              <TraceTimelineLink traceID={traceID} />
            </React.Fragment>
          ) : (
            <span className="u-tx-muted">Select a Trace...</span>
          )}
        </span>
        <IoChevronDown className="TraecDiffHeader--traceTitleChevron" />
      </h1>
      <AttrsComponent startTime={startTime} duration={duration} totalSpans={totalSpans} />
    </div>
  );
}

TraceHeader.defaultProps = {
  startTime: null,
  duration: null,
  error: undefined,
  state: undefined,
  totalSpans: null,
};
