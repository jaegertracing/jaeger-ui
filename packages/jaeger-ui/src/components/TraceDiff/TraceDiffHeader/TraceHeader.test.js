// Copyright (c) 2017 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import * as React from 'react';
import { shallow } from 'enzyme';

import TraceHeader, { Attrs, EmptyAttrs } from './TraceHeader';
import { fetchedState } from '../../../constants';

describe('TraceHeader', () => {
  const props = {
    duration: 700,
    error: { errorKey: 'errorValue' },
    traceID: 'trace-id',
    traceName: 'trace name',
  };
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<TraceHeader {...props} />);
  });

  it('renders as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders populated attrs component when props.state === fetchedState.DONE', () => {
    wrapper.setProps({
      startTime: 150,
      totalSpans: 50,
      state: fetchedState.DONE,
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders "Select a Trace..." when props.traceID is not provided ', () => {
    wrapper.setProps({
      traceID: null,
    });
    expect(wrapper.find('.u-tx-muted').text()).toBe('Select a Trace...');
  });

  describe('EmptyAttrs', () => {
    it('renders as expected', () => {
      expect(shallow(<EmptyAttrs />)).toMatchSnapshot();
    });
  });

  describe('Attrs', () => {
    it('renders as expected when provided props', () => {
      expect(shallow(<Attrs duration={700} startTime={150} totalSpans={50} />)).toMatchSnapshot();
    });

    it('Attrs renders as expected when missing props', () => {
      expect(shallow(<Attrs />)).toMatchSnapshot();
    });
  });
});
