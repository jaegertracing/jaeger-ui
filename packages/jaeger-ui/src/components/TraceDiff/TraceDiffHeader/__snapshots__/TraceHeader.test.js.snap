// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TraceHeader Attrs Attrs renders as expected when missing props 1`] = `
<ul
  className="TraecDiffHeader--traceAttributes"
>
  <li
    className="TraecDiffHeader--traceAttr"
  >
    <strong>
      <RelativeDate
        fullMonthName={true}
        includeTime={true}
        value={0}
      />
    </strong>
  </li>
  <li
    className="TraecDiffHeader--traceAttr"
  >
    <span
      className="u-tx-muted"
    >
      Duration: 
    </span>
    <strong>
      0ms
    </strong>
  </li>
  <li
    className="TraecDiffHeader--traceAttr"
  >
    <span
      className="u-tx-muted"
    >
      Spans: 
    </span>
     
    <strong>
      0
    </strong>
  </li>
</ul>
`;

exports[`TraceHeader Attrs renders as expected when provided props 1`] = `
<ul
  className="TraecDiffHeader--traceAttributes"
>
  <li
    className="TraecDiffHeader--traceAttr"
  >
    <strong>
      <RelativeDate
        fullMonthName={true}
        includeTime={true}
        value={0.15}
      />
    </strong>
  </li>
  <li
    className="TraecDiffHeader--traceAttr"
  >
    <span
      className="u-tx-muted"
    >
      Duration: 
    </span>
    <strong>
      0.7ms
    </strong>
  </li>
  <li
    className="TraecDiffHeader--traceAttr"
  >
    <span
      className="u-tx-muted"
    >
      Spans: 
    </span>
     
    <strong>
      50
    </strong>
  </li>
</ul>
`;

exports[`TraceHeader EmptyAttrs renders as expected 1`] = `
<ul
  className="TraecDiffHeader--traceAttributes"
>
  <li
    className="TraecDiffHeader--traceAttr"
  >
    Â 
  </li>
</ul>
`;

exports[`TraceHeader renders as expected 1`] = `
<div
  className="TraecDiffHeader--traceHeader"
>
  <h1
    className="TraecDiffHeader--traceTitle"
  >
    <span>
      <React.Fragment>
        <TraceName
          className=""
          error={
            Object {
              "errorKey": "errorValue",
            }
          }
          key="name"
          state={null}
          traceName="trace name"
        />
         
        <small
          className="u-tx-muted ub-pr2"
          key="id"
        >
          trace-i
        </small>
        <TraceTimelineLink
          traceID="trace-id"
        />
      </React.Fragment>
    </span>
    <IoChevronDown
      className="TraecDiffHeader--traceTitleChevron"
    />
  </h1>
  <EmptyAttrs
    duration={700}
    startTime={null}
    totalSpans={null}
  />
</div>
`;

exports[`TraceHeader renders populated attrs component when props.state === fetchedState.DONE 1`] = `
<div
  className="TraecDiffHeader--traceHeader"
>
  <h1
    className="TraecDiffHeader--traceTitle"
  >
    <span>
      <React.Fragment>
        <TraceName
          className=""
          error={
            Object {
              "errorKey": "errorValue",
            }
          }
          key="name"
          state="FETCH_DONE"
          traceName="trace name"
        />
         
        <small
          className="u-tx-muted ub-pr2"
          key="id"
        >
          trace-i
        </small>
        <TraceTimelineLink
          traceID="trace-id"
        />
      </React.Fragment>
    </span>
    <IoChevronDown
      className="TraecDiffHeader--traceTitleChevron"
    />
  </h1>
  <Attrs
    duration={700}
    startTime={150}
    totalSpans={50}
  />
</div>
`;
